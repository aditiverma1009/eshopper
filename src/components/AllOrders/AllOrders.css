.all-products {
  padding: 10px;
}
