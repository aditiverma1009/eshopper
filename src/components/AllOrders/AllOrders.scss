.all-products {
  padding: 10px;
  .all-products-table {
    border-collapse: collapse;
    margin: auto;

    .order-rows > td {
      border: 1px solid #dddddd;
      text-align: left;
      padding: 8px;
    }
  }
  .item-table {
    border: 0;
    width: 100%;
    border-collapse: collapse;
    margin-bottom: 5px;
    td {
      width: 100px;
      border-right: 1px solid #dddddd;
      padding: 0 10px;
    }
    td:first-child {
      width: 20px;
    }
    td:last-child {
      width: 20px;
      border: 0;
    }
  }
}
