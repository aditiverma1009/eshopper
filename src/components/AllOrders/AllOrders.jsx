import React, { Component } from "react";
import axios from "axios";
import BASE_URL from "../../constants/index";
import "./AllOrders.css";

export default class AllOrders extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      orders: [],
    };
  }

  componentDidMount = async () => {
    const { data, error } = await axios.get(`${BASE_URL}/orders`);
    const orders = data.data;
    if (orders) {
      this.setState({
        isLoaded: true,
        orders: orders,
      });
    } else if (error) {
      this.setState({
        isLoaded: true,
        error: error,
      });
    }
  };

  render() {
    const { error, isLoaded, orders } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    }
    return (
      <div className="all-products">
        {orders.map((eachOrder) => {
          return (
            <div key={eachOrder.id}>
              {eachOrder.id}--- {Date(eachOrder.date)}
            </div>
          );
        })}
      </div>
    );
  }
}
