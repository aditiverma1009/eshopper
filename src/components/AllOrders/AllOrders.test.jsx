import { render, waitFor } from "@testing-library/react";
import AllOrders from "./AllOrders.jsx";
import React from "react";
import axiosMock from "axios";

const mockOrders = {
  data: {
    data: [
      {
        items: [
          {
            id: 1,
            name: "apple",
            price: 120,
            count: 1,
            category: "Fruits & Vegatables",
          },
        ],
        id: 1,
        date: 1615122360481,
      },
    ],
  },
};

describe("All Orders", () => {
  test("should get orders", async () => {
    axiosMock.get.mockResolvedValueOnce(mockOrders);
    const { getByText } = render(<AllOrders />);
    const result = await waitFor(() => getByText("Order ID"));
    expect(result).toBeInTheDocument();
    expect(axiosMock.get).toHaveBeenCalledTimes(1);
    expect(axiosMock.get).toHaveBeenCalledWith("/orders");
  });
});
