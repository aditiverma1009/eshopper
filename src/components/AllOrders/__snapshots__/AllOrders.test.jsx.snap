// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Orders should get orders 1`] = `
<div>
  <div
    class="all-products"
  >
    <table
      class="all-products-table"
    >
      <tr
        class="order-rows"
      >
        <td>
          Order ID
        </td>
        <td>
          Date
        </td>
        <td>
          Items
        </td>
      </tr>
      <tr
        class="order-rows"
      >
        <td>
          1
        </td>
        <td>
          Sun Mar 07 2021 18:36:00 GMT+0530 (India Standard Time)
        </td>
        <td>
          <table
            class="item-table"
          >
            <tr>
              <td>
                1
              </td>
              <td>
                apple
              </td>
              <td>
                1
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  </div>
</div>
`;

exports[`All Orders should match snapshot for loading 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
